clusterprovider/kubeflex/kubeflex.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kflexprovider
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/klog/v2"
+
+	clusterprovider "github.com/kubestellar/kubestellar/pkg/clustermanager/providerclient"
+)
+
+// KflexClusterProvider is a kubeflex cluster provider
+type KflexClusterProvider struct {
+	providerName string
+	watch        clusterprovider.Watcher
+}
+
+// New creates a new KflexClusterProvider
+func New(providerName string) KflexClusterProvider {
+	return KflexClusterProvider{
+		providerName: providerName,
+	}
+}
+
+// TODO: switch from CLI to getting this info via kube directives
+func (k KflexClusterProvider) Create(name string, opts clusterprovider.Options) error {
+	logger := klog.Background()
+	logger.V(2).Info("Creating KubeFlex cluster", "name", name)
+	cmd := exec.Command("kflex", "create", name)
+	err := cmd.Run()
+
+	if err != nil {
+		// TODO:  Need to differentiate between "already exists" and an error
+		logger.Error(err, "Failed to create cluster", "name", name)
+	}
+
+	return err
+}
+
+// TODO: switch from CLI to getting this info via kube directives
+func (k KflexClusterProvider) Delete(name string, opts clusterprovider.Options) error {
+	logger := klog.Background()
+	logger.V(2).Info("Deleting KubeFlex cluster", "name", name)
+	cmd := exec.Command("kflex", "delete", name)
+	return cmd.Run()
```

Comment:
No error in this one?

Reply:
The err is returned on the cmd.Run() line 67. But you have a point.  This is running on a separate thread and won't necessarily be checked by the invoker.  I added an explicit check that dumps a probe.